packages/core/src/components/overlay/overlay.tsx

```diff
@@ -72,11 +72,26 @@ export interface IOverlayableProps {
     /**
      * A callback that is invoked when user interaction causes the overlay to close, such as
      * clicking on the overlay or pressing the `esc` key (if enabled).
+     *
      * Receives the event from the user's interaction, if there was an event (generally either a
      * mouse or key event). Note that, since this component is controlled by the `isOpen` prop, it
      * will not actually close itself until that prop becomes `false`.
      */
-    onClose?(event?: React.SyntheticEvent<HTMLElement>): void;
+    onClose?: (event?: React.SyntheticEvent<HTMLElement>) => void;
+}
+
+export interface IOverlayLifecycleProps {
+    /** Lifecycle method invoked when an Overlay begins to close. (Specifically, when the close transition begins.) */
+    onClosing?: () => void;
+
+    /** Lifecycle method invoked when an Overlay has finished transitioning to the closed state. */
+    onClosed?: () => void;
+
+    /** Lifecycle method invoked when an Overlay begins to open. */
```

Comment:
i think it might be clearer to say:
`Lifecycle method invoked just before an Overlay begins to open.`

Because we're dealing with transitions, "when" is ambiguous so we should say "before" or "after" the transition.

Same with `onClosing`